package maxtrack.pb.commands;

message u_parameter
{
  required string id                         = 1;
  required string value                      = 2;
}

message u_command
{
	required uint32 protocol                 = 1;
	required uint32 serial                   = 2;
	required string id                       = 3;
	required uint32 type                     = 4;
	required uint32 attempt                  = 5;
	required string timeout                  = 6;
	required uint32 handler_type             = 7;
	repeated u_parameter parameter           = 8;
	optional string transport                = 9; // Caso não seja informado a aplicação deverá usar gprs
	optional string satellite_terminal_id    = 10;
	optional uint32 peripheral               = 11; // 0 (Não periferico), 1 (td50), 2 1 (td40) se não informado a palicação deverá usar 0
	optional bool delayed                    = 12;
	optional uint32 library                  = 13;
	optional uint32 message_id               = 14;
	optional string satellite_account_id     = 15;
	optional string satellite_sub_account_id = 16;
	optional string next_command_id			 = 17;
	optional string satellite_provider		 = 18;
}

message u_command_response
{
  required uint32 protocol                   = 1;
  required uint32 serial                     = 2;
  required string id                         = 3;
  required uint32 handler_type               = 4;
  required uint32 fragment_number            = 5;
  required uint32 fragment_count             = 6;
  required uint32 status                     = 7;
  required string date                       = 8;
  required uint32 attempt                    = 9;
  optional uint32 message_id                 = 10;
  optional bool transport_answered		     = 11;
  required string description				 = 12;
} 

message u_command_pwd_gen_response
{
  required uint32 protocol                   = 1;
  required uint32 serial                     = 2;
  required string id                         = 3;
  required uint32 handler_type               = 4;
  required uint32 status                     = 5;
  required string date                       = 6;
  optional string password                   = 7;
  optional double latitude                   = 8;
  optional double longitude                  = 9;
  optional uint32 day                        = 10;
  optional uint32 hour                       = 11;
  optional uint32 minute                     = 12;
  optional uint32 alarm                      = 13;
  optional uint32 io_type                    = 14;
  optional uint32 io_number                  = 15;
}